.NET/Microsoft.Recognizers.Text.NumberWithUnit/Extractors/NumberWithUnitExtractor.cs

```diff
@@ -327,6 +349,27 @@ public void ExtractSeparateUnits(string source, List<ExtractResult> numDependRes
                                 matchResult[j] = true;
                             }
 
+                            //Special treatment, handle cases like '2:00 pm', both '00 pm' and 'pm' are not dimension
+                            var isDimensionFallsInTime = false;

```

Comment:
Why is this repeated here? If really needed, it should be a call to the same method.